### PR TITLE
Remove commented out code and refactor to rename variables appropriately

### PR DESCRIPTION
**Pull request by DavidHoltkamp1**
Removed some commented out code.
Renamed variables for better convention.

### CODE REVIEW

File: lib/game_teams.rb [hunk 1]
```diff
@@ -46,7 +46,6 @@ def self.worst_fans
   end
 
   def self.games_per_team
-    # @@games_per_team ||=
```
Comment (DavidHoltkamp1) [inline lib/game_teams.rb]: memoization?

File: lib/game_teams.rb [hunk 2]
```diff
@@ -83,51 +82,51 @@ def self.difference_home_and_away_wins_percentage
...
     Team.team_id_to_team_name(order[0])
   end
 
   def self.highest_scoring_visitor
-    order = away_games.max_by do |team_id, game_array|
-      total = game_array.sum {|game| game.goals.to_f}
-      (total / game_array.length).round(2)
+    order = away_games.max_by do |team_id, game_collection|
+      total = game_collection.sum {|game| game.goals.to_f}
+      (total / game_collection.length).round(2)
```
Comment (DavidHoltkamp1) [inline lib/game_teams.rb]: this is better convention when naming variables?

File: test/season_test.rb [hunk 3]
```diff
@@ -15,10 +15,6 @@ def test_it_exists
     assert_instance_of Season, @season
   end
 
-  def test_it_has_attributes
-    # assert_equal '20122013', @season.game_collection
```
Comment (DavidHoltkamp1) [inline test/season_test.rb]: Removed this commented out test

File: lib/game_teams.rb [hunk 4]
```diff
@@ -83,51 +82,51 @@ def self.difference_home_and_away_wins_percentage
 
   def self.away_games
     away_games = {}
-    games_per_team.each do |team_id, game_array|
-      away_games[team_id] = game_array.find_all {|game|game.hoa == 'away'}
+    games_per_team.each do |team_id, game_collection|
```
Comment (AlexisDumortier) [inline lib/game_teams.rb]: fixing naming of variable to avoid including the data type in the name

File: lib/game_teams.rb [hunk 5]
```diff
@@ -83,51 +82,51 @@ def self.difference_home_and_away_wins_percentage
...
     end
     order[0]
     Team.team_id_to_team_name(order[0])
   end
 
   def self.highest_scoring_visitor
-    order = away_games.max_by do |team_id, game_array|
-      total = game_array.sum {|game| game.goals.to_f}
-      (total / game_array.length).round(2)
+    order = away_games.max_by do |team_id, game_collection|
```
Comment (AlexisDumortier) [inline lib/game_teams.rb]: good catch!

**Pull request merged**